test/libponyc/sugar.cc

```diff
@@ -1479,173 +1479,6 @@ TEST_F(SugarTest, AsDontCareMultiTuple)
-    "  new val create(): $T val^ => true";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectWithField)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object let x: T = 3 fun foo() => 4 end";
-
-  const char* full_form =
-    "use \"builtin\"\n"
-    "class ref Foo\n"
-    "  var create: U32\n"
-    "  fun box f(): None =>\n"
-    "    $T.create(3)\n"
-    "    None\n"
-
-    "class ref $T\n"
-    "  let x: T\n"
-    "  fun box foo(): None =>\n"
-    "    4\n"
-    "    None\n"
-    "  new ref create($1: T): $T ref^ => x = consume $1";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectWithBehaviour)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object be foo() => 4 end";
-
-  const char* full_form =
-    "use \"builtin\"\n"
-    "class ref Foo\n"
-    "  var create: U32\n"
-    "  fun box f(): None =>\n"
-    "    $T.create()\n"
-    "    None\n"
-
-    "actor tag $T\n"
-    "  be tag foo(): None =>\n"
-    "    4\n"
-    "  new tag create(): $T tag^ => true";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectBox)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object box fun foo() => 4 end";
-
-  const char* full_form =
-    "use \"builtin\"\n"
-    "class ref Foo\n"
-    "  var create: U32\n"
-    "  fun box f(): None =>\n"
-    "    $T.create()\n"
-    "    None\n"
-
-    "primitive val $T\n"
-    "  fun box foo(): None =>\n"
-    "    4\n"
-    "    None\n"
-    "  new val create(): $T val^ => true";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectRef)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object ref fun foo() => 4 end";
-
-  const char* full_form =
-    "use \"builtin\"\n"
-    "class ref Foo\n"
-    "  var create: U32\n"
-    "  fun box f(): None =>\n"
-    "    $T.create()\n"
-    "    None\n"
-
-    "class ref $T\n"
-    "  fun box foo(): None =>\n"
-    "    4\n"
-    "    None\n"
-    "  new ref create(): $T ref^ => true";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectTagWithBehaviour)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object tag be foo() => 4 end";
-
-  const char* full_form =
-    "use \"builtin\"\n"
-    "class ref Foo\n"
-    "  var create: U32\n"
-    "  fun box f(): None =>\n"
-    "    $T.create()\n"
-    "    None\n"
-
-    "actor tag $T\n"
-    "  be tag foo(): None =>\n"
-    "    4\n"
-    "  new tag create(): $T tag^ => true";
-
-  TEST_EQUIV(short_form, full_form);
-}
-
-
-TEST_F(SugarTest, ObjectRefWithBehaviour)
-{
-  const char* short_form =
-    "class Foo\n"
-    "  var create: U32\n"
-    "  fun f() =>\n"
-    "    object ref be foo() => 4 end";
-
-  TEST_ERROR(short_form);
-}
```

Comment:
There's a lot of tests removed here that were covering some important aspects of object literals.

I understand that the object processing was moved out of the `sugar` pass and into the `expr` pass, so that these tests don't really belong in this file anymore, and that `TEST_EQUIV` no longer makes sense.  However, adapting these tests for the `expr` pass and figuring out how to cover the same aspects would be ideal, so we can prevent regressions as we move forward.

Also, any other tests that can be made to reflect possible pitfalls of the capture process would be great.  I saw that this PR adds a test to `builtin_test`, but it doesn't seem like enough to really cover this major feature, especially with all these tests having been removed.

If you don't have the time to devote to adding more tests to this PR, I'd be happy to help chip in and make sure we have some more tests.